test/integration/certificates/generates_new_private_key_per_request_test.go

```diff
@@ -0,0 +1,344 @@
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	controllerpkg "github.com/cert-manager/cert-manager/pkg/controller"
+	"github.com/cert-manager/cert-manager/pkg/controller/certificates"
+	"github.com/cert-manager/cert-manager/pkg/controller/certificates/issuing"
+	"github.com/cert-manager/cert-manager/pkg/controller/certificates/keymanager"
+	"github.com/cert-manager/cert-manager/pkg/controller/certificates/readiness"
+	"github.com/cert-manager/cert-manager/pkg/controller/certificates/requestmanager"
+	"github.com/cert-manager/cert-manager/pkg/controller/certificates/revisionmanager"
+	"github.com/cert-manager/cert-manager/pkg/controller/certificates/trigger"
+	testpkg "github.com/cert-manager/cert-manager/pkg/controller/test"
+	logf "github.com/cert-manager/cert-manager/pkg/logs"
+	"github.com/cert-manager/cert-manager/pkg/metrics"
+	"github.com/cert-manager/cert-manager/pkg/util/pki"
+	"github.com/cert-manager/cert-manager/test/integration/framework"
+)
+
+func TestGeneratesNewPrivateKeyIfMarkedInvalidRequest(t *testing.T) {
+	namespace := "default"
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
+	config, stopFn := framework.RunControlPlane(t, ctx)
+	defer stopFn()
+
+	// Build, instantiate and run all required controllers
+	stopControllers := runAllControllers(t, ctx, config)
+	defer stopControllers()
+
+	_, _, cmCl, _ := framework.NewClients(t, config)
+	crt, err := cmCl.CertmanagerV1().Certificates(namespace).Create(ctx, &cmapi.Certificate{
+		ObjectMeta: metav1.ObjectMeta{Name: "testcrt"},
+		Spec: cmapi.CertificateSpec{
+			SecretName: "testsecret",
+			DNSNames:   []string{"something"},
+			IssuerRef: cmmeta.ObjectReference{
+				Name: "issuer",
+			},
+			PrivateKey: &cmapi.CertificatePrivateKey{
+				// This doesn't actually make any difference in this test case because there is no existing private
+				// key, meaning there's no private key to re-use.
+				RotationPolicy: cmapi.RotationPolicyAlways,
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("failed to create certificate: %v", err)
+	}
+
+	var firstReq *cmapi.CertificateRequest
+	if err := wait.Poll(time.Millisecond*500, time.Second*10, func() (done bool, err error) {
+		reqs, err := cmCl.CertmanagerV1().CertificateRequests(namespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		if len(reqs.Items) > 1 {
+			return false, fmt.Errorf("invalid state, expected only one CR but got %d", len(reqs.Items))
+		}
+
+		if len(reqs.Items) == 0 {
+			return false, nil
+		}
+
+		firstReq = &reqs.Items[0]
+		return true, nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("Found CertificateRequest")
+	// Remember the CSR data used for the first request so we can compare it later
+	originalCSR := firstReq.Spec.Request
+
+	// Mark the CSR as 'Failed'
```

Comment:
**question (blocking):** This comment should say `'InvalidRequest'`, right? As opposed to `'Failed'` which is true for the second test? It confused me at first trying to figure out what the difference was between this and the second test